File: hint/src/main/java/datacomprojects/com/hint/Tip.java
package datacomprojects.com.hint;

import android.content.Context;
import android.view.View;

import androidx.annotation.Nullable;

public class Tip {
    public boolean isShow = false;
    public String hintID;
    public TipView customHint;

    public Tip(String hintID, TipView customHint, @Nullable View view, @Nullable TipsList list) {
        this.hintID = hintID;
        this.customHint = customHint;
        if(view!=null)
            this.customHint.attachToView(view);
    }

    public boolean showIfNeed(Context context) {
        if(!TipsSharedPreferencesUtils.getInstance(context).getBoolean(hintID,false)) {
            customHint.show();
            isShow = true;
            return true;
        }
        return false;
    }

    public boolean wasShowed(Context context) {
        return TipsSharedPreferencesUtils.getInstance(context).getBoolean(hintID,false);
    }

    public void hide(Context context) {
        TipsSharedPreferencesUtils.getInstance(context).putBoolean(hintID,true).apply();
        customHint.hide();
        isShow = false;
    }

    public void reset(Context context) {
        TipsSharedPreferencesUtils.getInstance(context).putBoolean(hintID,false).apply();
    }
}

File: hint/src/main/java/datacomprojects/com/hint/Tip.kt
package datacomprojects.com.hint

import android.content.Context
import android.graphics.drawable.Drawable
import android.view.View

class Tip @JvmOverloads constructor(var hintID: String, var customHint: TipView, view: View?, closeDrawable: Drawable? = null) {
    var isShow = false

    init {
        view?.let { this.customHint.attachToView(it) }
        closeDrawable?.let { customHint.setCloseImage(it) }
    }

    fun showIfNeed(context: Context, onShowed: () -> Unit) {
        if (!TipsSharedPreferencesUtils.getInstance(context).getBoolean(hintID, false)) {
            customHint.show(onShowed)
            isShow = true
        }
    }

    fun wasShowed(context: Context): Boolean {
        return TipsSharedPreferencesUtils.getInstance(context).getBoolean(hintID, false)
    }

    fun hide(context: Context, onAnimationEnd: (() -> Unit)? = null) {
        TipsSharedPreferencesUtils.getInstance(context).putBoolean(hintID, true).apply()
        customHint.hide(onAnimationEnd)
        isShow = false
    }

    fun reset(context: Context) {
        TipsSharedPreferencesUtils.getInstance(context).putBoolean(hintID, false).apply()
    }
}
